File: 卡码/代码随想录/Go/贪心/01-简单/LeetCode_0455.go
//455. 分发饼干
package main

import "sort"

func findContentChildren(g []int, s []int) int {
	sort.Ints(g)
	sort.Ints(s)
	index:=len(s)-1
	res:=0
	for i:=len(g)-1;i>=0;i-- {
		if index<0 {
			break
		}
		if s[index]>=g[i] {
			res++
			index--
		}
	}
	return res
}

File: 卡码/代码随想录/1/08-贪心算法/Java/greedy/src/main/java/com/kurumi/greedy/_01_简单/LeetCode_0455.java
//455. 分发饼干
package com.kurumi.greedy._01_简单;

import java.util.Arrays;

public class LeetCode_0455 {

}
class Solution {
  public int findContentChildren(int[] g, int[] s) {
    Arrays.sort(g);
    Arrays.sort(s);
    int index=s.length-1;
    int res=0;
    for(int i=g.length-1;i>=0;i--){
      if(index<0) break;
      if(s[index] >= g[i]){
        index--;
        res++;
      }
    }
    return res;
  }
}
